lib/services/sf.js

```diff
@@ -1121,6 +1121,12 @@ StateConnecting.prototype.continue = function ()
       this.connectionConfig.getJsTreatIntegerAsBigInt();
   }
 
+  if (Util.exists(this.connectionConfig.getGcsUseDownscopedCredential()))
```

Comment:
I can see token pass, but where is the code to use GCS SDK for get/put instead of HTTPS call?

Reply:
Currently, the connector doesn't use the GCS SDK so it makes HTTPS calls with the GCS access token in the authorization header: https://github.com/snowflakedb/snowflake-connector-nodejs/blob/ea8db10873f1c38b3aeed84fa7b418af4a7730b4/lib/file_transfer_agent/gcs_util.js#L232

Similar to the python connector: https://github.com/snowflakedb/snowflake-connector-python/blob/main/src/snowflake/connector/gcs_storage_client.py#L156

Reply:
I'll update the PR to use the GCS SDK instead when the access token is available

Reply:
Please follow the JDBC code changes. As Python somehow going away from SDK.

Reply:
If you want to follow Python and not to add any dependency then that's fine also. As Python, Node.js classify as connectors rather than drivers.